wpe/src/main/cpp/browser/page.cpp

```diff
@@ -46,6 +46,17 @@ void onTitleChanged(WebKitWebView* webView, GParamSpec*, gpointer data)
         observer->onTitleChanged(title, canGoBack, canGoForward);
     }
 }
+
+bool onDomFullscreenRequest(void* data, bool fullscreen)
+{
+    ALOGV("onDomFullscreenRequest() fullscreen: %s", fullscreen?"true":"false");
+    auto* page = reinterpret_cast<Page*>(data);
+    if (page != nullptr) {
```

Comment:
I find the first one (in 2 lines) more readable. But this is a question of taste I suppose.
In the one line version I suppose you should add something like:
```
if ((auto* page = reinterpret_cast<Page*>(data)) != nullptr) {
```
if you don't want complains with static analyzers like clang-tidy.
That could be a good idea to add this tool, it helps a lot to enforce good practices in C++

Reply:
Doing `if (pointer)` is the same as `if (pointer != nullptr)`, clang-tidy should not warn about that. The WebKit code has plenty of variable declaration + assignment + check in `if` statements and I am quite confident that the extra parens and the explicit `nullptr` check are not needed at all; they are not ambiguous since C++11 onwards.

Reply:
follows the convention that was already used elsewhere in the file for similar code. We can reformat whole file later with consistent convention